### PR TITLE
Add RFP 

### PR DESCRIPTION
**Pull request by PolyProgrammist**
## Summary by CodeRabbit

- **New Features**
  - Enhanced proposal management with new functionalities for creating and querying proposals and RFPs.
  - Added capabilities to create dumps and snapshots for proposals and RFPs.
  - Improved querying features to fetch the latest views and snapshots for proposals and RFPs.

### CODE REVIEW

File: proposals/index.js [hunk 1]
```diff
@@ -12,22 +12,85 @@ import { Block } from "@near-lake/primitives";
...
+    }
   }
 }
 
+function getAddOrEditObject(block, startsWith) {
+  const stateChanges = block.streamerMessage.shards
+    .flatMap((e) => e.stateChanges)
+    .filter(
+      (stateChange) =>
+        stateChange.change.accountId === "truedove38.near" &&
```
Comment (frol) [inline proposals/index.js]: Don't forget to update the account back to `devhub.near` (here and in some other places)

**Review by Tguntenaar — COMMENTED** [review]
Looks good to me! 

You might want to take this into account or we put it in a different issue:

The indexer sometimeess fails to write data due to a constraint violation.

That is because we are using the original way of writing data to the indexer using GraphQL mutation. This generally works, but provides less flexibility and has poor performance. I recommend migrating to [context.db methods](https://docs.near.org/bos/queryapi/context-object#db-methods). Try to use upsert extensively to allow your indexer to reindex same blocks.

So instead of await context.graphql we can use upsert.

**Comment by PolyProgrammist** [comment]
Hi @Tguntenaar  

I switched to using contextdb. So it should be more optimal, as Pavel exlained:
```
You could win in performance in a big way when you move away from GraphQL simply because 1) we won’t spend time parsing GraphQL and translating it to SQL and 2) we won’t need to spend time on HTTPS handshakes and ecnryption/decription etc
```

I didn't use upserts though. My thoughts about upserts are the following:
 - Upsert stands for "insert or update". We don't have updates. We just have inserts. And it's more straightforward to use just insert. 
 - We can also use upserts for the purpose of re-running the indexer on the same blocks if we want to fix a bug. However, I find it pretty simple to just delete all the data and rerun from the start. It doesn't take too much time. Besides, if we use upsert for that purpose, we have to specify `updateColumns` (the last argument for upsert) which is error-prone

**Review by Tguntenaar — APPROVED** [review]
Great job!

**Comment by PolyProgrammist** [comment]
Just in case:

I use the UI of dataplatform.near to debug. I use GraphQL tab to delete data and then just insert the js code and publish it again. It requires more steps when you change the schema though

Query to delete data:
```
mutation deleteDumb($receipt_id: polyprogrammist_near_devhub_proposals_sierra_dumps_updates! = "hey") {
  delete_polyprogrammist_near_devhub_objects_s_rfp_dumps(where: {receipt_id: {_neq: "pepe"}}) {
    affected_rows
  }
  delete_polyprogrammist_near_devhub_objects_s_rfp_snapshots(where: {}) {
    affected_rows
  }
 delete_polyprogrammist_near_devhub_objects_s_rfps(where: {id: {_neq: 500}}) {
    affected_rows
  }
  delete_polyprogrammist_near_devhub_objects_s_dumps(where: {receipt_id: {_neq: "pepe"}}) {
    affected_rows
  }
  delete_polyprogrammist_near_devhub_objects_s_proposals(where: {id: {_neq: 500}}) {
    affected_rows
  }
  delete_polyprogrammist_near_devhub_objects_s_proposal_snapshots(where: {}) {
    affected_rows
  }
}
```

**Pull request merged**